### PR TITLE
add new keyboard hannah60rgb

### PR DESCRIPTION
**Pull request by mechlovin**
## Description

add hannah60rgb pcb, gh60 and unikorn compatible

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/mechlovin/hannah60rgb/hannah60rgb.c [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  1,          1,          1,          1,          1,          1,          1,          1,          1,          1,          1,          1, 
+} };
+
+void rgb_matrix_indicators_user(void)
+{
+        if (IS_HOST_LED_ON(USB_LED_CAPS_LOCK))
+        {
+            rgb_matrix_set_color(30, 255, 255, 255);
+        }
+}
```
Comment (drashna) [inline keyboards/mechlovin/hannah60rgb/hannah60rgb.c:46]: This is mostly a nit pick, but ...

```suggestion
void rgb_matrix_indicators_user(void) {
        if (IS_HOST_LED_ON(USB_LED_CAPS_LOCK)) {
            rgb_matrix_set_color(30, 255, 255, 255);
        }
}
```

File: keyboards/mechlovin/hannah60rgb/keymaps/default/config.h [hunk 2]
```diff
@@ -0,0 +1,19 @@
+/* Copyright 2020 Mechlovin'
```
Comment (fauxpark) [inline keyboards/mechlovin/hannah60rgb/keymaps/default/config.h:1]: This file can be deleted.

File: keyboards/mechlovin/hannah60rgb/hannah60rgb.c (hunk 1 above)
Comment (mechlovin) [reply keyboards/mechlovin/hannah60rgb/hannah60rgb.c:46]: > This is mostly a nit pick, but ...

Thank you. It looks much neater now

File: keyboards/mechlovin/hannah60rgb/hannah60rgb.c [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "hannah60rgb.h"
+led_config_t g_led_config = { {
```
Comment (drashna) [inline keyboards/mechlovin/hannah60rgb/hannah60rgb.c:18]:
```suggestion

#ifdef RGB_MATRIX_ENABLE
led_config_t g_led_config = { {
```

File: keyboards/mechlovin/hannah60rgb/hannah60rgb.c [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  {  0, 64},  { 16, 64},  { 32,64 },  { 48,64 },  { 64,64},   { 80,64},   { 96,64},   { 112,64},  { 144,64},  { 160,64},  { 192,64},  { 224,64},                         
+}, {
+  1,          1,          1,          1,          1,          1,          1,          1,          1,          1,          1,          1,            1,           1,           1,
+  1,          1,          1,          1,          1,          1,          1,          1,          1,          1,          1,          1,            1,           1,           1,  
+  1,          1,          1,          1,          1,          1,          1,          1,          1,          1,          1,          1,            1,           1,           1,  
+  1,          1,          1,          1,          1,          1,          1,          1,          1,          1,          1,          1,            1,           1,           1,  
+  1,          1,          1,          1,          1,          1,          1,          1,          1,          1,          1,          1, 
+} };
```
Comment (drashna) [inline keyboards/mechlovin/hannah60rgb/hannah60rgb.c:38]:
```suggestion
} };
#endif
```

File: keyboards/mechlovin/hannah60rgb/hannah60rgb.c [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+#endif
+
+void rgb_matrix_indicators_user(void) {
+        if (IS_HOST_LED_ON(USB_LED_CAPS_LOCK)) {
+            rgb_matrix_set_color(30, 255, 255, 255);
+        }
+}
```
Comment (zvecr) [inline keyboards/mechlovin/hannah60rgb/hannah60rgb.c:47]:
```suggestion

void rgb_matrix_indicators_user(void) {
        if (IS_HOST_LED_ON(USB_LED_CAPS_LOCK)) {
            rgb_matrix_set_color(30, 255, 255, 255);
        }
}
#endif
```

File: keyboards/mechlovin/hannah60rgb/hannah60rgb.c [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+}, {
+  1,          1,          1,          1,          1,          1,          1,          1,          1,          1,          1,          1,            1,           1,           1,
+  1,          1,          1,          1,          1,          1,          1,          1,          1,          1,          1,          1,            1,           1,           1,  
+  1,          1,          1,          1,          1,          1,          1,          1,          1,          1,          1,          1,            1,           1,           1,  
+  1,          1,          1,          1,          1,          1,          1,          1,          1,          1,          1,          1,            1,           1,           1,  
+  1,          1,          1,          1,          1,          1,          1,          1,          1,          1,          1,          1, 
+} };
+
+void rgb_matrix_indicators_user(void) {
+        if (IS_HOST_LED_ON(USB_LED_CAPS_LOCK)) {
```
Comment (fauxpark) [inline keyboards/mechlovin/hannah60rgb/hannah60rgb.c:43]:
```suggestion
        if (get_host_led_state().caps_lock) {
```

File: keyboards/mechlovin/hannah60rgb/rules.mk [hunk 7]
```diff
@@ -0,0 +1,28 @@
+# MCU name
+MCU = STM32F303
+
+
```
Comment (fauxpark) [inline keyboards/mechlovin/hannah60rgb/rules.mk:4]:
```suggestion
```

**Comment by mechlovin** [comment]
I just updated according to what I understand.
Am I doing it right?
Thank you! :D

File: keyboards/mechlovin/hannah60rgb/hannah60rgb.h [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+K30, K31, K32, K33, K34, K35, K36, K37, K38, K39, K3A, K3B, K3C, K3D,      \
+K40, K41, K42,      K44,      K46,      K48,      K4A, K4B, K4C, K4D       \
+) { \
+    { K00,   K01,   K02,   K03,   K04,   K05,   K06,   K07,   K08,   K09,   K0A,   K0B,   K0C,   K0D   }, \
+    { K10,   K11,   K12,   K13,   K14,   K15,   K16,   K17,   K18,   K19,   K1A,   K1B,   K1C,   K1D   }, \
+    { K20,   K21,   K22,   K23,   K24,   K25,   K26,   K27,   K28,   K29,   K2A,   K2B,   K2C,   K2D   }, \
+    { K30,   K31,   K32,   K33,   K34,   K35,   K36,   K37,   K38,   K39,   K3A,   K3B,   K3C,   K3D   }, \
+    { K40,   K41,   K42,   KC_NO, K44,   KC_NO, K46,   KC_NO, K48,   KC_NO, K4A,   K4B,   K4C,   K4D   }, \
+}
+#define LAYOUT_ansi( \
```
Comment (noroadsleft) [inline keyboards/mechlovin/hannah60rgb/hannah60rgb.h:34]:
```suggestion
#define LAYOUT_60_ansi( \
```

File: keyboards/mechlovin/hannah60rgb/hannah60rgb.h [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+                 K30,      K32, K33, K34, K35, K36, K37, K38, K39, K3A, K3B, K3C,      \
+                 K40, K41, K42,                K46,                K4A, K4B, K4C, K4D  \
+) { \
+    { K00,   K01,   K02,   K03,   K04,   K05,   K06,   K07,   K08,   K09,   K0A,   K0B,   K0C,   K0D   }, \
+    { K10,   K11,   K12,   K13,   K14,   K15,   K16,   K17,   K18,   K19,   K1A,   K1B,   K1C, KC_NO   }, \
+    { K20,   K21,   K22,   K23,   K24,   K25,   K26,   K27,   K28,   K29,   K2A,   K2B,   K2C,   K2D   }, \
+    { K30, KC_NO,   K32,   K33,   K34,   K35,   K36,   K37,   K38,   K39,   K3A,   K3B,   K3C, KC_NO   }, \
+    { K40,   K41,   K42, KC_NO, KC_NO,   KC_NO, K46,   KC_NO, KC_NO, KC_NO, K4A,   K4B,   K4C,   K4D   }, \
+}
+#define LAYOUT_iso( \
```
Comment (noroadsleft) [inline keyboards/mechlovin/hannah60rgb/hannah60rgb.h:47]:
```suggestion
#define LAYOUT_60_iso( \
```

File: keyboards/mechlovin/hannah60rgb/info.json [hunk 10]
```diff
@@ -0,0 +1,81 @@
+{
+    "keyboard_name": "hannah60rgb",
+    "url": "",
+    "maintainer": "qmk",
+    "width": 15,
+    "height": 5,
+    "layouts": {
+        "LAYOUT": {
```
Comment (noroadsleft) [inline keyboards/mechlovin/hannah60rgb/info.json:8]:
```suggestion
        "LAYOUT_all": {
```

File: keyboards/mechlovin/hannah60rgb/readme.md [hunk 11]
```diff
@@ -0,0 +1,14 @@
+# hannah60rgb
+
+![hannah60rgb](https://i.imgur.com/ImXgsyX.png)
```
Comment (noroadsleft) [inline keyboards/mechlovin/hannah60rgb/readme.md:3]:
```suggestion
![hannah60rgb](https://i.imgur.com/ImXgsyXl.png)
```

File: keyboards/mechlovin/hannah60rgb/readme.md [hunk 12]
```diff
@@ -0,0 +1,14 @@
...
+
+60% PCB standard, Unikorn keyboard compatible.
```
Comment (noroadsleft) [inline keyboards/mechlovin/hannah60rgb/readme.md:5]: If I'm understanding the Hardware Supported bit correctly...

```suggestion
A 60% PCB with per-key RGB, compatible with Poker and Unikorn cases.
```

File: keyboards/mechlovin/hannah60rgb/readme.md [hunk 13]
```diff
@@ -0,0 +1,14 @@
...
+
+* Keyboard Maintainer: [Mechlovin'](https://github.com/mechlovin)
+* Hardware Supported: Unikorn, standard pocker 60% PCB
```
Comment (noroadsleft) [inline keyboards/mechlovin/hannah60rgb/readme.md:8]:
```suggestion
* Hardware Supported: hannah60rgb PCB, STM32F303
```

File: keyboards/mechlovin/hannah60rgb/readme.md [hunk 14]
```diff
@@ -0,0 +1,14 @@
...
+
+Make example for this keyboard (after setting up your build environment):
+
+    make hannah60rgb:default
```
Comment (noroadsleft) [inline keyboards/mechlovin/hannah60rgb/readme.md:12]:
```suggestion
    make mechlovin/hannah60rgb:default
```

**Comment by noroadsleft** [comment]
Also, where do I get one of these PCBs? Asking for... myself. :smiley:

**Comment by mechlovin** [comment]
oh, thank you for help me!
Just pm my partner Flying#1179 on discord. Thank you so much 🥰

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**